### DISCUSSION TITLE
How is the Focus Handling Done for the `MultipleChoiceFilter`?

### DISCUSSION THREAD

**Discussion by neurolag** [dotnet/DataGridExtensions#99] [Q&A]
I tried to create my own implementation of the `FilterWithPopupControl` from the examples.
However, even when trying to implement the `FilterWithPopupControl` identically to the `MultipleChoiceFilter`, I get unexpected results:

When clicking on the `ToggleButton` of the `MultipleChoiceFilter`, the application still responds to my mouse hovering other controls.

However, when trying to do the same with my own `FilterWithPopupControl`, anything outside the popup won't respond to hovering.

I even copied the `PopupFocusManagerBehavior` to my own project which didn't change the result.

How can I create my own filter with popup to work the same like `MultipleChoiceFilter` does?

Sadly I don't have a repo I could link to as it is a company related project.

**Comment by tom-englert** [comment]
Hard to say without knowing what you are doing

**Reply by neurolag** [reply]
I'll go give it a try to create an isolated example to upload to GitHub

Thank you so much for your answer

**Reply by neurolag** [reply]
Thank you so much for your patience.
I finally managed it to create a rough example here:
https://github.com/manuth/DGXPopupDemo

Notice the difference in the behavior of the built-in `MultiChoiceFilter` and the handmade `NumericFilterControl`:
* When opening the built-in `MultiChoiceFilter`
  * The first item in the popup is getting focussed (take note that the first checkbox is triggered when hitting the space bar)
  * Both the `MainWindow` and the popup respond to hovering: Both DataGrid rows and mail addresses are still getting highlighted when hovering
* When opening the handmade `NumericFilterControl`:
  * Much like in the built-in `MultiChoiceFilter` control, the first item in the popup is focussed
  * Items in the `MainWindow` do **not** response to hovering. Only items in the popup react to hovering.

Do you know what's causing this difference?
